models/Import/Service/SendCalculatedResultService.php

```diff
@@ -85,28 +128,51 @@ public function sendByDeliveryExecutionId(string $deliveryExecutionId): void
                 $scoreTotalMax = (float)$variable->getValue();
             }
         }
-
-        $this->eventManager->trigger(
-            new DeliveryExecutionResultsRecalculated(
-                $deliveryExecution,
-                $scoreTotal,
-                $scoreTotalMax
-            )
-        );
+        return [$scoreTotal, $scoreTotalMax];
     }
 
-    /**
-     * @throws common_exception_Error
-     * @throws InvalidServiceManagerException
-     */
-    private function getResultsStorage(): ReadableResultStorage
+    private function getGradingStatus(string $deliveryExecutionId, array $outcomeVariables): string
     {
-        $storage = $this->resultServerService->getResultStorage();
-
-        if (!$storage instanceof ReadableResultStorage) {
-            throw new common_exception_Error('Configured result storage is not writable.');
+        $qtiTestItems = $this->qtiTestItemsService->getItemsByDeliveryExecutionId($deliveryExecutionId);
+        $gradingStatus = ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED;
+
+        foreach ($qtiTestItems as $parts) {
```

Comment:
1 foreach instead of 4
```suggestion
        foreach ($testDeclarations as $declarationAttributes) {
          if (!isset($declarationAttributes['externalScored'])) {
            continue;
           }
           $gradingStatus = ScoreInterface::GRADING_PROGRESS_STATUS_PENDING_MANUAL;
           if($this->isSubjectOutcomeVariableGraded(
               $outcomeVariables,
               $declarationAttributes['identifier']
             ) {
               $gradingStatus = ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED;
            }
....... 
```

Reply:
Done